### PR TITLE
fix: proposed token important fix

### PR DESCRIPTION
**Pull request by shellscape**
**Which issue #** if any, does this resolve?
#118 

Please check one:
- [x] New tests created for this change
- [x] Tests updated for this change

This PR:
- [ ] Adds new API
- [ ] Extends existing API, backwards-compatible
- [ ] Introduces a breaking change
- [x] Fixes a bug

---

From #124
> This should be a better approach to it than #119, the logic is mostly from core `postcss`. I'm hoping to find a way to consolidate it to one solution across both libraries.

This PR supersedes #124. It uses a different approach which alters the tokenizer when `important` identifiers are split between multiple tokens.

### CODE REVIEW

File: lib/LessParser.js [hunk 1]
```diff
@@ -114,6 +113,29 @@ module.exports = class LessParser extends Parser {
         tokens = tokens.concat(tokensAfter);
       }
 
+      const importantTokens = [];
+
+      for (const token of tokens) {
```
[the remark this answers is not in the document]
Comment (shellscape) [reply lib/LessParser.js]: hm but when the space is before the `!`, it _should_ go into `raws.between`

**Comment by shellscape** [comment]
@jwilsson updated those two tests

File: lib/LessParser.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwilsson) [reply lib/LessParser.js]: I guess it's a question if we wanna follow the way PostCSS does for `Decl` with `!important`, there the spaces go into `raws.important` but perhaps it makes more sense to put it into `raws.between` for `AtRule`s? You decide :)

[the remark this answers is not in the document]
Comment (shellscape) [reply lib/LessParser.js]: yeah, I considered the `Decl` implementation. that's a slightly different node type with different rules for parsing and stringifying though - at least, that's what it looked like to me :) With the mixins it struck me that the `!important` wasn't so much a decorator as a component part of the mixin, and while using `raws.important` works for some consistency, tracking spaces in `raws.between` makes for easier inspection and composition, and it's less prone to errors (from what I can recall).

[the remark this answers is not in the document]
Comment (jwilsson) [reply lib/LessParser.js]: Having slept on it, I think I agree :) They are different node types as and long as it's documented I think it's alright.

**Pull request merged**